src/ssl_stubs.c

```diff
@@ -446,28 +420,24 @@ static const SSL_METHOD *get_method(int protocol, int type)
       caml_invalid_argument("Unknown method (this should not have happened, please report).");
       break;
   }
-  caml_acquire_runtime_system();
-
-  if (method == NULL)
-    caml_raise_constant(*caml_named_value("ssl_exn_method_error"));
-
-  return method;
 }
 
 CAMLprim value ocaml_ssl_create_context(value protocol, value type)
 {
   CAMLparam2(protocol, type);
   CAMLlocal1(block);
   SSL_CTX *ctx;
-  const SSL_METHOD *method = get_method(Int_val(protocol), Int_val(type));
+  const SSL_METHOD *method = get_method(Int_val(type));
```

Comment:
I noticed you converted this double locking/unlocking to a single one.

I reverted it in this change to keep it more faithful to the original (and easier to review), but will submit it in a follow-up PR.